src/InstagramScraper/Instagram.php

```diff
@@ -387,6 +389,63 @@ public function getMediaCommentsByCode($code, $count = 10, $maxId = null)
         }
         return $comments;
     }
+	
+	/**
+     * @param      $code
+     * @param int  $count
+     * @param null $maxId
+     *
+     * @return array
+     * @throws InstagramException
+     */
+    public function getMediaLikesByCode($code, $count = 10, $maxId = null)
+    {
+        $remain = $count;
+        $likes = [];
+        $index = 0;
+        $hasPrevious = true;
+        while ($hasPrevious && $index < $count) {
+            if ($remain > self::MAX_LIKES_PER_REQUEST) {
+                $numberOfLikesToRetreive = self::MAX_LIKES_PER_REQUEST;
+                $remain -= self::MAX_LIKES_PER_REQUEST;
+                $index += self::MAX_LIKES_PER_REQUEST;
+            } else {
+                $numberOfLikesToRetreive = $remain;
+                $index += $remain;
+                $remain = 0;
+            }
+            if (!isset($maxId)) {
+                $maxId = '';
+
+            }
+            $commentsUrl = Endpoints::getLastLikesByCode($code, $numberOfLikesToRetreive, $maxId);
+            $response = Request::get($commentsUrl, $this->generateHeaders($this->userSession));
+            if ($response->code !== 200) {
+                throw new InstagramException('Response code is ' . $response->code . '. Body: ' . $response->body . ' Something went wrong. Please report issue.');
+            }
+            $cookies = self::parseCookies($response->headers['Set-Cookie']);
+            $this->userSession['csrftoken'] = $cookies['csrftoken'];
+            $jsonResponse = json_decode($response->raw_body, true);
+            
+			$nodes = $jsonResponse['data']['shortcode_media']['edge_liked_by']['edges'];
+			
+            foreach ($nodes as $likesArray) {
+                $likes[] = Like::create($likesArray['node']);
+            }
+			
+            $hasPrevious = $jsonResponse['data']['shortcode_media']['edge_liked_by']['page_info']['has_next_page'];
+            $numberOfLikes = $jsonResponse['data']['shortcode_media']['edge_liked_by']['count'];
+            if ($count > $numberOfLikes) {
+                $count = $numberOfLikes;
+            }
+            if (sizeof($nodes) == 0) {
+                return $likes;
+            }
+            $maxId = $nodes[sizeof($nodes) - 1]['node']['id'];
```

Comment:
This doesn't work for likes (it won't fetch a second batch if $count is more than MAX_LIKES_PER_REQUEST). Instead, use

$maxId = $jsonResponse['data']['shortcode_media']['edge_liked_by']['page_info']['end_cursor'];

Reply:
Spot on that fixed it thanks!

Reply:
One question, I set the max likes per batch to 300 to equal the same as comments but what would be a better max batch number to use?

Reply:
@eversionsystems for now it is okay in my opinion. After release we will get feedback if something is wrong.